app/controllers/comments_controller.rb

```diff
@@ -11,11 +11,33 @@ def edit
   def create
     @comment = Comment.new(comment_params)
     @comment.user = current_user
-
     respond_to do |format|
       if @comment.save
+        reference = @comment.root.commentable
+
+        if @comment.parent
+          parent_id = @comment.parent_id
+          new_comment = @comment.parent.children.build
+          display_comment = false
+        else
+          parent_id = "r#{reference.id}"
+          new_comment = reference.comments.build
+          display_comment = true
+        end
+
         format.html { redirect_to :back, notice: 'Comment was successfully created.' }
         format.json { render :show, status: :created, location: @comment }
+        format.js {
+          render(
+            'create.js.erb',
+            locals: {
+              reference: reference,
+              parent_id: parent_id,
+              new_comment: new_comment,
+              display_comment: display_comment
```

Comment:
I could go either way on this issue, but this seems like a lot of locals to be cramming into one partial.